test/run_pass/test_net_10.js

```diff
@@ -39,6 +41,11 @@ socket1.on('connect', function() {
 });
 
 process.on('exit', function() {
-  assert(timedout);
+  if (process.platform == 'nuttx' || process.platform == 'tizenrt') {
```

Comment:
The problem is that connect is synchronous on NUTTX. Perhaps we can invent something for this, but the test must be changed (hopefully in a platform independent way).